.github/workflows/nodejs_bindings.yml

```diff
@@ -0,0 +1,319 @@
+name: Build Node.js bindings
+
+on:
+  # We run this once a week so we regularly publish weekly builds
+  # but not on _every_ master commit
+  schedule:
+    - cron: "0 0 * * 0"
+  push:
+    paths-ignore:
+      - "**.md"
+      - .circleci/
+      - docs/
+      - run_route_scripts/
+      - test_requests/
+      - test/check_compilers/
+      - .github/workflows/check_compilers.yml
+      - ".tx"
+      - ".git*"
+    branches:
+      - master
+    tags:
+      - '*'
+  pull_request:
+    paths-ignore:
+      - "*.md"
+      - .circleci/
+      - docs/
+      - run_route_scripts/
+      - test_requests/
+      - test/check_compilers/
+      - .github/workflows/check_compilers.yml
+      - ".tx"
+      - ".git*"
+    branches:
+      - master
+
+# Cancel CI runs on a branch when new commits were pushed
+# See https://stackoverflow.com/a/72408109/2582935
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build_nodejs_bindings:
+    name: Build Node.js bindings (${{ matrix.os }}-${{ matrix.arch }})
+    runs-on: ${{ matrix.runner }}
+    container: ${{ matrix.container }}
+    strategy:
+      matrix:
+        include:
+          - os: linux
+            arch: x64
+            runner: ubuntu-latest
+            container: ghcr.io/valhalla/manylinux:2_28_valhalla_python
+          - os: linux
+            arch: arm64
+            runner: ubuntu-24.04-arm
+            container: ghcr.io/valhalla/manylinux:2_28_valhalla_python
+          - os: macos
+            arch: arm64
+            runner: macos-14
+            container: null
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-depth: 0
+          fetch-tags: true
+      
+      - name: Setup ccache
+        uses: actions/cache@v4
+        with:
+          path: ${{ matrix.os == 'macos' && '/Users/runner/Library/Caches/ccache' || '.ccache' }}
+          key: ccache-${{ matrix.os }}-${{ matrix.arch }}-${{ github.sha }}
+          restore-keys: |
+            ccache-${{ matrix.os }}-${{ matrix.arch }}-
+
+      - name: Install dependencies (macOS)
+        if: matrix.os == 'macos'
+        run: |
+          HOMEBREW_NO_AUTO_UPDATE=1 brew install boost protobuf libspatialite luajit spatialite-tools dylibbundler
+      - name: Build Valhalla with Node.js bindings
+        run: |
+          ${{ matrix.os == 'linux' && 'export LD_LIBRARY_PATH="/usr/local/lib64:/usr/local/lib"' || '' }}
+          # since we are running in a container, we need to set the ccache directory to folder accessible from GHA runner
```

Comment:
did ccache work for you on linux? I think on the pyhton ones it's not really working at this point